src/backends/policy_engine/souffle/check_policy_compliance.cc

```diff
@@ -72,36 +119,48 @@ int main(int argc, char* argv[]) {
   google::InitGoogleLogging("check_policy_compliance");
   absl::SetProgramUsageMessage(kUsageMessage);
   absl::ParseCommandLine(argc, argv);
-  // Parse the given IR file.
-  absl::StatusOr<std::string> ir_string =
-      ReadFileContents(absl::GetFlag(FLAGS_ir));
-  if (!ir_string.ok()) {
-    LOG(ERROR) << "Error reading IR file: " << ir_string.status();
-    return 2;
-  }
 
   // Read the sql policy rules file.
   absl::StatusOr<std::string> sql_policy_rules =
       ReadFileContents(absl::GetFlag(FLAGS_sql_policy_rules).value());
-  if (absl::GetFlag(FLAGS_sql_policy_rules).has_value() &&
-      !sql_policy_rules.ok()) {
+  if (absl::GetFlag(FLAGS_sql_policy_rules).has_value() && !sql_policy_rules.ok()) {
     LOG(ERROR) << "Error reading sql policy rules file: "
                << sql_policy_rules.status();
-    return 2;
+    return UnwrapExitCode(ReturnCode::ERROR);
+  }
+
+  auto ir_path = absl::GetFlag(FLAGS_ir);
+  auto proto_path = absl::GetFlag(FLAGS_proto);
```

Comment:
[`auto` should only be used when the type is obvious or when there's a substantial readability benefit.](https://google.github.io/styleguide/cppguide.html#Type_deduction)